iothub/service/src/DirectMethod/DirectMethodsClient.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Devices.Common.Exceptions;
+using Microsoft.Azure.Devices.Http2;
+
+namespace Microsoft.Azure.Devices
+{
+    /// <summary>
+    /// Subclient of <see cref="IotHubServiceClient"/> to invoke direct methods on devices and modules in IoT hub.
+    /// </summary>
+    /// <seealso href="https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-direct-methods"/>
+    public class DirectMethodsClient
+    {
+        private readonly string _hostName;
+        private readonly IotHubConnectionProperties _credentialProvider;
+        private readonly HttpClient _httpClient;
+        private readonly HttpRequestMessageFactory _httpRequestMessageFactory;
+
+        private const string DeviceMethodUriFormat = "/twins/{0}/methods";
+        private const string ModuleMethodUriFormat = "/twins/{0}/modules/{1}/methods";
+
+        /// <summary>
+        /// Creates an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        protected DirectMethodsClient()
+        {
+        }
+
+        internal DirectMethodsClient(string hostName, IotHubConnectionProperties credentialProvider, HttpClient httpClient, HttpRequestMessageFactory httpRequestMessageFactory)
+        {
+            _hostName = hostName;
+            _credentialProvider = credentialProvider;
+            _httpClient = httpClient;
+            _httpRequestMessageFactory = httpRequestMessageFactory;
+        }
+
+        /// <summary>
+        /// Invokes a method on a device.
+        /// </summary>
+        /// <param name="deviceId">The device identifier for the target device.</param>
+        /// <param name="cloudToDeviceMethod">Parameters to execute a direct method on the device.</param>
+        /// <param name="cancellationToken">A cancellation token to cancel the operation.</param>
+        /// <returns>The <see cref="CloudToDeviceMethodResult"/>.</returns>
```

Comment:
The web documentation will automatically link to the type. You don't need to include it here. If you do, the page ends up with more links and it just ends up being distracting.